src/azure/storage/credential.rs

```diff
@@ -26,11 +28,9 @@ impl Credential {
             return false;
         }
         if let Credential::BearerToken(_, expires_on) = self {
-            if let Ok(expires) = chrono::DateTime::parse_from_rfc3339(expires_on) {
-                let buffer = chrono::Duration::try_minutes(2).expect("in bounds");
-                if expires > (chrono::Utc::now() + buffer) {
-                    return false;
-                }
+            let buffer = chrono::TimeDelta::try_minutes(2).expect("in bounds");
+            if expires_on > &(chrono::Utc::now() + buffer) {
```

Comment:
Is this correct?

Reply:
Let's wait for the longevity test result.